### PR TITLE
[webview_flutter] Migrate deprecated Scaffold snackbars

### PR DESCRIPTION
**Pull request by Piinks**
This migrates the deprecated Scaffold methods to ScaffoldMessenger in example code from the webview plugin.
These are being removed in https://github.com/flutter/flutter/pull/98549
Part of https://github.com/flutter/flutter/issues/98537

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran [the auto-formatter]. (Unlike the flutter/flutter repo, the flutter/plugins repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[the auto-formatter]: https://github.com/flutter/plugins/blob/master/script/tool/README.md#format-code
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

File: packages/webview_flutter/webview_flutter/example/lib/main.dart [hunk 1]
```diff
@@ -152,19 +151,21 @@ class _WebViewExampleState extends State<WebViewExample> {
...
+            onPressed: () async {
+              String? url;
+              if (controller.hasData) {
+                url = (await controller.data!.currentUrl())!;
+              }
+              ScaffoldMessenger.of(context).showSnackBar(
+                SnackBar(
+                    content: Text(
+                  controller.hasData ? 'Favorited $url' : 'Unable to favorite',
+                )),
```
Comment (goderbauer) [inline packages/webview_flutter/webview_flutter/example/lib/main.dart:164]: formatting here is odd, maybe just needs a trailing `,` between the `))`?

File: packages/webview_flutter/webview_flutter/example/test/main_test.dart [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:webview_flutter/webview_flutter.dart';
+import 'package:webview_flutter_example/main.dart';
+
+void main() {
+  testWidgets('Test snackbar from ScaffoldMessenger',
+      (WidgetTester tester) async {
+    await tester.pumpWidget(
+        MaterialApp(home: WebViewExample(cookieManager: FakeCookieManager())));
```
Comment (goderbauer) [inline packages/webview_flutter/webview_flutter/example/test/main_test.dart:14]: Maybe sprinkle in some trailing commas to format this nicer?

**Review by goderbauer — APPROVED** [review]
LGTM

File: packages/webview_flutter/webview_flutter/CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,6 @@
-## NEXT
+## 3.0.2
 
+* Migrate deprecated `Scaffold.showSnackBar` to `ScaffoldMessenger`.
```
Comment (bparrishMines) [inline packages/webview_flutter/webview_flutter/CHANGELOG.md:3]: nit: `Migrate` should be `Migrates` to follow https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style. Also include that the change is only in the example app. This should also be done for `webview_flutter_android` and `webview_flutter_wkwebview`.

```suggestion
* Migrates deprecated `Scaffold.showSnackBar` to `ScaffoldMessenger` in example app.
```

**Review by bparrishMines — APPROVED** [review]
LGTM with a nit

**Pull request merged**